pintracker/optracker/operation.go

```diff
@@ -160,16 +172,32 @@ func (op *Operation) Error() string {
 // an error message. It updates the timestamp.
 func (op *Operation) SetError(err error) {
 	_, span := trace.StartSpan(op.ctx, "optracker/SetError")
+
+	prevStatus := op.ToTrackerStatus()
+
 	op.mu.Lock()
 	{
 		op.phase = PhaseError
 		op.error = err.Error()
 		op.ts = time.Now()
 	}
 	op.mu.Unlock()
+
+	op.recordStatuses(prevStatus, op.ToTrackerStatus())
 	span.End()
 }
 
+func (op *Operation) recordStatuses(prevStatus api.TrackerStatus, newStatus api.TrackerStatus) {
+	if !prevStatus.Match(newStatus) {
```

Comment:
Isn't it better to do `prevStatus == newStatus` ? Using match implies we expect a status may match several other statuses, but here we are just checking for equality right?